src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -124,29 +121,29 @@ public static SystemVersion GetSystemVersion()
             };
         }
 
-        private static void DetectRHEL6()
+        private static void DetectRHELVersion(string version)
         {
-            lock (detectedRHEL6lock)
+            lock (detectedRHELVersionlock)
             {
                 if (!RunningOnLinux || !File.Exists("/etc/redhat-release"))
                 {
-                    detectedRHEL6 = false;
+                    detectedRHELVersion = false;
```

Comment:
>detectedRHELVersion [](http://example.com/codeflow?start=20&length=19)

[maybe not for this PR], having this function change static members as a side effect not a good design, we should consider refactoring